transformation/plugins/org.jboss.tools.fuse.transformation.editor/src/main/java/org/jboss/tools/fuse/transformation/editor/internal/wizards/Hl7Page.java

```diff
@@ -0,0 +1,570 @@
+import org.eclipse.jface.layout.GridDataFactory;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.layout.FillLayout;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Event;
+import org.eclipse.swt.widgets.Group;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.swt.widgets.Text;
+import org.jboss.tools.fuse.transformation.editor.Activator;
+import org.jboss.tools.fuse.transformation.editor.internal.util.ClasspathResourceSelectionDialog;
+import org.jboss.tools.fuse.transformation.editor.internal.util.CompoundValidator;
+import org.jboss.tools.fuse.transformation.editor.internal.util.Util;
+
+/**
+ *
+ */
+public class Hl7Page extends XformWizardPage implements TransformationTypePage {
+
+    private Composite _page;
+    private boolean isSource = true;
+    private Text hl7FileText;
+    private Button hl7SchemaOption;
+    private Button hl7InstanceOption;
+    private Text hl7PreviewText;
+    private Binding _binding;
+
+    /**
+     * @param pageName
+     * @param model
+     * @param isSource
+     */
+    public Hl7Page(final String pageName, final Model model, final boolean isSource) {
+        super(pageName, model);
+        setTitle("HL7 Page");
+        setImageDescriptor(Activator.imageDescriptor("transform.png"));
+        this.isSource = isSource;
+        observablesManager.addObservablesFromContext(context, true, true);
+    }
+
+    private void bindControls() {
+
+        // Bind source file path widget to UI model
+        final IObservableValue widgetValue = WidgetProperties.text(SWT.Modify).observe(hl7FileText);
+        IObservableValue modelValue = null;
+        if (isSourcePage()) {
+            modelValue = BeanProperties.value(Model.class, "sourceFilePath").observe(model);
+        } else {
+            modelValue = BeanProperties.value(Model.class, "targetFilePath").observe(model);
+        }
+        final UpdateValueStrategy strategy = new UpdateValueStrategy();
+
+        PathValidator pathValidator = new PathValidator();
+        FileEmptyValidator fileEmptyValidator = new FileEmptyValidator();
+        JSONValidator jsonValidator = new JSONValidator();
+        CompoundValidator compoundJSONTextValidator = new CompoundValidator(
+                pathValidator, fileEmptyValidator, jsonValidator);
+        strategy.setBeforeSetValidator(compoundJSONTextValidator);
+
+        _binding = context.bindValue(widgetValue, modelValue, strategy, null);
+        ControlDecorationSupport.create(_binding, decoratorPosition, hl7FileText.getParent());
+
+        widgetValue.addValueChangeListener(new IValueChangeListener() {
+
+			@Override
+			public void handleValueChange(ValueChangeEvent event) {
```

Comment:
this is quite a big anonymous class, it might worth extracting it as a dedicated class